projects/plugins/jetpack/modules/related-posts/jetpack-related-posts.php

```diff
@@ -492,27 +493,35 @@ public function render_block( $attributes, $content ) {
 			$wrapper_attributes = \WP_Block_Supports::get_instance()->apply_block_supports();
 		}
 
-		$headline_id     = 'relatedposts-headline-' . uniqid();
 		$headline_markup = '';
 
-		if ( $block_attributes['show_headline'] === true ) {
+		if ( isset( $block ) ) {
+			foreach ( $block->inner_blocks as $inner_block ) {
+				if ( 'core/heading' === $inner_block->name && ! empty( trim( wp_strip_all_tags( $inner_block->inner_html ) ) ) ) {
+					$headline_markup = $inner_block->inner_html;
+					break;
+				}
+			}
+		}
+
+		if ( empty( $headline_markup ) && $block_attributes['show_headline'] === true ) {
 			$headline = $block_attributes['headline'];
 			if ( strlen( trim( $headline ) ) !== 0 ) {
 				$headline_markup = sprintf(
-					'<h3 id="%2$s" class="jp-relatedposts-headline">%1$s</h3>',
-					esc_html( $headline ),
-					esc_attr( $headline_id )
```

Comment:
I take it the `$headline_id` wasn't necessary anymore? Could this be an issue if folks had added links to that anchor somehow?

Reply:
I should have added a note here, too. See the previous comment for why I removed it. It's not an issue since the id was hardcoded: it's not user-populated.